### PR TITLE
[ROCM] added a cudnn switch of conv2d for rocm platform

### PR DESCRIPTION
**Pull request by ronny1996**
### PR types

Others
### PR changes

Others
### Describe

added a cudnn switch of conv2d for rocm platform

### CODE REVIEW

File: python/paddle/nn/layer/conv.py [hunk 1]
```diff
@@ -629,6 +633,11 @@ def __init__(self,
                  weight_attr=None,
                  bias_attr=None,
                  data_format="NCHW"):
+        use_cudnn = True
```
Comment (qili93) [inline python/paddle/nn/layer/conv.py]: 这里写成类似这样的形式
use_cudnn  = True if (core.is_compiled_with_cuda() and
                                   cudnn_version is not None) else False

File: python/paddle/nn/layer/conv.py [hunk 2]
```diff
@@ -153,6 +156,7 @@ def _get_default_param_initializer():
                                           out_channels % in_channels == 0):
             self._op_type = 'depthwise_conv2d'
             self._use_cudnn = False
+        self._use_cudnn = self._use_cudnn and use_cudnn
```
Comment (qili93) [inline python/paddle/nn/layer/conv.py]: 加一行注释，解释一下这里这么修改的原因，例如
// add environment of ROCM_CONV2D_DISABLE_CUDNN to disable use_cudnn in Conv2D
// If use_cudnn is not defined, default value is True, then self._use_cudnn = self._use_cudnn

**Review by qili93 — DISMISSED** [review]
LGTM, api test will be covered by CPU CI and CUDA CI

File: python/paddle/fluid/layers/nn.py [hunk 3]
```diff
@@ -1498,6 +1498,9 @@ def conv2d(input,
           conv2d = paddle.static.nn.conv2d(input=data, num_filters=2, filter_size=3, act="relu")
           print(conv2d.shape) # [-1, 2, 30, 30]
     """
+    use_cudnn = False if (
+        core.is_compiled_with_rocm() and
+        bool(os.getenv("ROCM_CONV2D_DISABLE_CUDNN"))) else use_cudnn
```
Comment (zhiqiu) [inline python/paddle/fluid/layers/nn.py]: If ROCM_CONV2D_DISABLE_CUDNN is not a built-in variable in ROCM but created by Paddle, I suggest following the naming rule like other environments like "FLAGS_xxx_xxx".

Comment (zhiqiu) [reply python/paddle/fluid/layers/nn.py]: Plus, it's better to get the value by `fluid.get_flags` instead of `os.getenv`

Comment (ronny1996) [reply python/paddle/fluid/layers/nn.py]: update

**Review by qili93 — DISMISSED** [review]
LGTM

**Review by zhiqiu — APPROVED** [review]
LGTM for paddle.fluid.layers.conv2d

**Pull request merged**